.travis.yml

```diff
@@ -1,14 +1,18 @@
 language: node_js
 node_js:
   - "node"
+
+script:
+  - npm install -g firebase-tools
+  - gulp inject
+
 deploy:
-  provider: s3
-  access_key_id: AKIAIRWAFHZDDWJ37X5A
-  secret_access_key:
-    secure: cScQWz2z+W5z+aK5tT6TMxp8vhIuWlcw4HQT1I3v8OLW6pHaAfPUyChWdfdKXV1itUK9Qt+YkQmCKGvOdYHtijB2xlBYQZJih/2+U7h/I52YbSW617QwGd3A1Jk4/Fym3LwlKcOCUOm7RDTscHipRjZQhGEvF7f1v2oYBdU81Bp3xdNugbjtJHLCi8Jq/9K8uKk8WD7bBj6UGhjK3JJJHJB6F41BpKhnltNAOTi+QtJShsmD25iJfYvyig3VP2fp5a6vvOKX7doKD0AduJm62CoeP65eUCHelN08pNuCI50YjBkkv+9hQ/Y7WnsOzkWjrEgUiNiA9FuhugO2Lvp6UioCk4RkzZY53QMdeFIgxoxZdhEz1wQgJIQ6fxTn3elGqGFCejZe0vMB5DPkK/2kqqqb501IvQf0wmptSt1pT/VfxZmrk2i718SKTdQluaIsKVGZyip6L49kjrlNq/uGqXcMwR4c85o239oGGWyzuWFSWWMPSqnpXScbRVFOzn+pp3Ei8ho5SfaI+N8tO/1/R+9vShtWeLQB7DQ23q1JDja+enNWhBuYshe6rkfb5zZzDy5+kOirCIGsK1V4+elmyIZHmue5xS7tZ7Hc/uXpuJXXGH/qnWglRwkt+ALnv/gGOXvR5+jhXeIadITIJqGqsRa3FzQupq67Pf+sdRbLgpw=
-  bucket: speed-measurementlab-net
+# production
+- provider: script
   skip_cleanup: true
-  local_dir: app
+  script: "firebase use mlab-oti && firebase deploy --only hosting:mlab-speedtest --token $FIREBASE_TOKEN"
   on:
-    repo: opentechinstitute/mlab-speedtest
-    branch: master
+    repo: m-lab/mlab-speedtest
+    all_branches: true
+    condition: "$TRAVIS_BRANCH == master"
```

Comment:
I don't think you need either `all_branches` or `condition`. Tags are not necessarily related to branches, and since we really only want to switch on whether a tag was applied, we can remove all conditions that have to do with branches.